### PR TITLE
Avoid passing default authorization header to workspace client

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

This PR should fix the issue when an access token is cached by the workspace client. 

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/19307

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
[dashboard-next-ci-test]

**Comment by dmytro-ndp** [comment]
[dashboard-next-ci-test]

**Comment by artaleks9** [comment]
@akurinnoy, ysterday I tested Che with this specific image and couldn't reproduce the bug described in the eclipse/che#19307. Stopping workspace works properly.

**Comment by akurinnoy** [comment]
@artaleks9 Thanks for the feedback!

File: src/services/workspace-client/index.ts [hunk 1]
```diff
@@ -85,15 +79,17 @@ export abstract class WorkspaceClient {
     if (keycloak) {
       return new Promise((resolve, reject) => {
         keycloak.updateToken(minValidity).success((refreshed: boolean) => {
-          if (refreshed && keycloak.token) {
-            const header = 'Authorization';
-            this.axios.defaults.headers.common[header] = `Bearer ${keycloak.token}`;
+          console.log('debug: refreshToken, minValidity', minValidity, ', config', config, ', refreshed', refreshed);
```
Comment (sleshchenko) [inline src/services/workspace-client/index.ts:82]: it should be removed before merging, right?

**Review by sleshchenko — APPROVED** [review]

File: src/services/workspace-client/index.ts (hunk 1 above)
Comment (akurinnoy) [reply src/services/workspace-client/index.ts:82]: yes, @artaleks9 hasn't reproduced the issue so I'll get rid of these outputs.

**Comment by sleshchenko** [comment]
[dashboard-next-ci-test]

**Review by JPinkney — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
[dashboard-next-ci-test]

**Comment by artaleks9** [comment]
@akurinnoy, on the last built image I checked stopping workspace on dashboard UI - it works properly.
Also I've launched E2E test to check stopping workspace from opened IDE - it works too.

**Pull request merged**